### DISCUSSION TITLE
How to completely remove/uninstall RustDesk: Rustdesk-Server?

### DISCUSSION THREAD

**Discussion by davidwebbespiritu** [rustdesk/rustdesk#1504] [Q&A]
Hello! I installed RustDesk Server using simple to run install script but now I want to completely remove/uninstall it from my server.
Anyone knows how to do it? Seems no guide for it yet. Thanks!

**Comment by corymar** [comment]
Hi!

This is not applicable for Docker-based RustDesk installations

I've installed RustDesk 1.1.6 on Ubuntu 22.04 Server following installation manual with install.sh script from here:
https://rustdesk.com/docs/en/self-host/install/#how-to-install-the-server

When I decided to remove it also found no uninstaller. So that's what I did:
```
sudo systemctl stop rustdeskrelay.service rustdesksignal.service
sudo systemctl disable rustdeskrelay.service rustdesksignal.service
sudo rm -rf /var/log/rustdesk/
sudo rm -rf /opt/rustdesk/
sudo rm -rf /etc/systemd/system/rustdesksignal.service
sudo rm -rf /etc/systemd/system/rustdeskrelay.service

sudo systemctl stop gohttpserver.service
sudo systemctl disable gohttpserver.service
sudo rm -rf /var/log/gohttp/
sudo rm -rf /opt/gohttp/
sudo rm -rf /etc/systemd/system/gohttpserver.service
```

If you have gohttpserver used by other applications or you didn't install it during RustDesk setup just don't execute commands with gohttp

**Reply by plao** [reply]
Thanks!

**Reply by PapaZivi** [reply]
THX

**Reply by WeskerC** [reply]
really helpful！

**Comment by remoover** [comment]
Hello,

can anyone tell me how to completely purge rustdesk from server ? is there an script or ... ? i am not able remove it ...

**Reply by SergeyMy** [reply]
server or client? server is higher.

**Reply by remoover** [reply]
I need remove server completly

**Reply by dinger1986** [reply]
How did you install it?

**Reply by remoover** [reply]
via script https://rustdesk.com/docs/en/self-host/pro/installscript/

**Reply by SergeyMy** [reply]
if native then:
```
systemctl status rustdesk-hbbr
systemctl status rustdesk-hbbs
#in order to find out the paths to the binaries
#then everything is simple
sudo systemctl stop rustdesk-hbbr
sudo systemctl stop rustdesk-hbbs
```
...
```
sudo systemctl disable rustdesk-hbbr
sudo systemctl disable rustdesk-hbbs
sudo rm /usr/bin/hbbr
sudo rm /usr/bin/hbbs
sudo rm -rf /var/lib/rustdesk-server
sudo rm -rf  /var/log/rustdesk-server
```
```
sudo rm  /etc/systemd/system/rustdesk-hbbs.service
sudo rm  /etc/systemd/system/rustdesk-hbbr.service
sudo systemctl daemon-reload
```
if rustdesknginx
```
sudo rm /etc/nginx/sites-available/rustdesk.conf
sudo rm /etc/nginx/sites-enabled/rustdesk.conf
```

**Reply by dinger1986** [reply]
So the opposite of the script 

```
sudo systemctl stop rustdesk-hbbs.service
sudo systemctl disable rustdesk-hbbs.service
sudo systemctl stop rustdesk-hbbr.service
sudo systemctl disable rustdesk-hbbr.service
sudo rm /etc/systemd/system/rustdesk-hbbs.service
sudo rm etc/systemd/system/rustdesk-hbbr.service
sudo rm /usr/bin/hbbs
sudo rm /usr/bin/hbbr
sudo rm -rf /var/lib/rustdesk-server/
```

if nginx was installed
`sudo apt remove nginx`

**Reply by dinger1986** [reply]
Only do nginx if you aren’t using it for anything else

**Reply by SergeyMy** [reply]
_Only do nginx if you aren’t using it for anything else_
well at least you warned me! :):):):):)

**Reply by remoover** [reply]
yes its removed now, many thanks !!!! :))) This will be fine to add into documentation...

**Reply by SergeyMy** [reply]
add, everything is in your hands, rustdesk.com/docs is also open for editing by the communities.

**Comment by SergeyMy** [comment]
@dinger1986
> So the opposite of the script

mine is fuller :) without

`sudo systemctl daemon-reload`

 there will be unintended consequences

**Reply by dinger1986** [reply]
Lol not fair @SergeyMy you edited yours, but yeah you are right! 😂. Tbf mine was written from my phone

**Comment by xlionjuan** [comment]
Build it on Docker next time

**Reply by SergeyMy** [reply]
Docker for hbbs+hbbr IMHO redundant
hbbs+hbbr is easy to assemble natively even on the i386 platform
![изображение](https://github.com/rustdesk/rustdesk/assets/131688106/9f0bb563-4e95-4c03-8bcc-28765739b1a8)

**Comment by imonlygitting** [comment]
one of the PCs keeps reporting errors (others are fine) so need to uninstall completely on windows, is there a way to do that? Tried to uninstall and reinstall but errors persisting so probably need a clean uninstall to fix the problem...

any suggestions most appreciated!
